tools/accuracy_checker/accuracy_checker/data_readers/data_reader.py

```diff
@@ -396,11 +401,14 @@ def read(self, data_id):
         if field_id is not None:
             key = [k for k, v in self.keyRegex.items() if v.match(field_id)]
             if len(key) > 0:
-                recno = field_id.split('_')[-1]
-                recno = int(recno)
-                start, _ = Path(data_id).name.split('.')
-                start = int(start)
-                return data[key[0]][recno - start, :]
+                if self.block:
+                    return data[key[0]]
+                else:
+                    recno = field_id.split('_')[-1]
+                    recno = int(recno)
+                    start, _ = Path(data_id).name.split('.')
```

Comment:
it is not safe for case when file have several suffixes
```suggestion
                    start = Path(data_id).name.split('.')[0]
```